File: backend/app/schemas/company_settings.py
import uuid

from pydantic import BaseModel, ConfigDict


class CompanySettingsOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    tenant_id: uuid.UUID
    store_id: uuid.UUID
    legal_name: str
    display_name: str
    address: str
    gstin: str | None
    fssai_no: str | None
    phone: str | None
    logo_url: str | None
    invoice_footer_text: str | None


class CompanySettingsUpdate(BaseModel):
    legal_name: str | None = None
    display_name: str | None = None
    address: str | None = None
    gstin: str | None = None
    fssai_no: str | None = None
    phone: str | None = None
    invoice_footer_text: str | None = None


class LogoUploadResponse(BaseModel):
    logo_url: str

File: frontend/src/api/companySettings.ts
import { apiClient } from "@/api/client";
import type { CompanySettings } from "@/types/bill";

export interface CompanySettingsUpdate {
  legal_name?: string;
  display_name?: string;
  address?: string;
  pincode?: string;
  gstin?: string | null;
  fssai_no?: string | null;
  phone?: string | null;
  invoice_footer_text?: string | null;
  show_tamil_item_names?: boolean;
}

export async function getCompanySettings(storeId?: string): Promise<CompanySettings> {
  const { data } = await apiClient.get<CompanySettings>("/settings/company", {
    params: { store_id: storeId },
  });
  return data;
}

export async function updateCompanySettings(
  payload: CompanySettingsUpdate,
  storeId?: string,
): Promise<CompanySettings> {
  const { data } = await apiClient.patch<CompanySettings>("/settings/company", payload, {
    params: { store_id: storeId },
  });
  return data;
}

export async function uploadCompanyLogo(file: File, storeId?: string): Promise<{ logo_url: string }> {
  const formData = new FormData();
  formData.append("file", file);
  const { data } = await apiClient.post<{ logo_url: string }>(
    "/settings/company/logo",
    formData,
    { params: { store_id: storeId }, headers: { "Content-Type": "multipart/form-data" } },
  );
  return data;
}
